src/main/java/net/sf/rails/ui/swing/AutoLoadPoller.java

```diff
@@ -13,91 +13,102 @@
 
 
 public class AutoLoadPoller extends Thread {
-    
-    private GameUIManager guiMgr;
+
+    private final GameUIManager guiMgr;
     private String saveDirectory;
     private String savePrefix;
     private String ownPostfix;
     private int pollingInterval;
     private int pollingStatus;
-    
+
     private boolean pollingActive = false;
-    
-    private String lastSavedFilenameFilepath;
+
+    private final String lastSavedFilenameFilepath;
     private String lastSavedFilename = "";
-    
+
     public static final int OFF = 0;
     public static final int ON = 1;
     public static final int SUSPENDED = 2;
 
-    protected static Logger log =
-        LoggerFactory.getLogger(AutoLoadPoller.class);
+    protected static Logger log = LoggerFactory.getLogger(AutoLoadPoller.class);
 
     public AutoLoadPoller (GameUIManager guiMgr, String saveDirectory, String savePrefix, String ownPostfix,
             int status, int pollingInterval) {
- 
         this.guiMgr = guiMgr;
         this.saveDirectory = saveDirectory;
         this.savePrefix = savePrefix;
         this.ownPostfix = ownPostfix;
         this.pollingStatus = status;
         this.pollingInterval = pollingInterval;
-        
+
         lastSavedFilenameFilepath = saveDirectory + "/" + savePrefix + ".last_rails";
-        
-        log.debug("Poller own postfix: "+ownPostfix);
-        log.debug("Poller last-filename path: "+lastSavedFilenameFilepath);
-        
+
+        log.debug("Poller own postfix: {}", ownPostfix);
+        log.debug("Poller last-filename path: {}", lastSavedFilenameFilepath);
     }
 
     @Override
     public void run () {
-
         log.info ("AutoLoadPoller started");
 
+        int currentPollInterval = 1;
         int secs, sleepTime;
-        String currentFilename;
 
         for (;;) {
```

Comment:
Off-topic: should we use `for (;;)` or `while (true)`? Or can we maybe use some kind of scheduler?

Reply:
Personally I never ever use `for (;;)`, I always use `while (true)` although realistically we should be checking a flag and shutting the poller down on exit so this case shouldn't really be either of these!